corehq/util/dates.py

```diff
@@ -18,3 +18,28 @@ def get_timestamp(date):
 
 def get_timestamp_millis(date):
     return 1000 * get_timestamp(date)
+
+
+def safe_strftime(val, fmt):
```

Comment:
Oops this was supposed to have been moved from another file, but I messed up the commit.

The main point of it is to avoid the following error in an (albeit edge) case where you're passing in a pre-1900 date/time:

```python
In [1]: import datetime

In [2]: datetime.date(1899, 1, 1).strftime('%Y')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-db7a1f50dcad> in <module>()
----> 1 datetime.date(1899, 1, 1).strftime('%Y')

ValueError: year=1899 is before 1900; the datetime strftime() methods require year >= 1900
```